pycbc/workflow/pegasus_workflow.py

```diff
@@ -788,6 +790,15 @@ def set_subworkflow_properties(self, output_map_file,
 
         self.add_planner_arg('pegasus.dir.storage.mapper.replica.file',
                              os.path.basename(output_map_file.name))
+        # Ensure output_map_file has the for_planning flag set. There's no
+        # API way to set this after the File is initialized, so we have to
+        # change the attribute here.
+        # WORSE, we only want to set this if the pegasus *planner* is version
+        # 5.0.4 or larger
+        sproc_out = subprocess.check_output(['pegasus-version']).strip()
+        sproc_out = sproc_out.decode()
+        if version.parse(sproc_out) >= version.parse('5.0.4'):
+            output_map_file.for_planning=True
```

Comment:
I note that it looks like pegasus is now forcing this flag for the dax file itself in 5.0.6 .... There's no obvious way to avoid this now, so this looks to be a version incompatibility between 5.0.6 and 5.0.3. It might then be worth removing this complexity, just set `output_map_file.for_planning=True` and require `pegasus-wms` (not the api) version to be 5.0.6 or larger. .... Perhaps this line does help as it avoids a failure if using pegasus 5.0.3 (consistently between the API and the executables).